### PR TITLE
Release update to pysat

### PR DESCRIPTION
**Pull request by rstoneback**
Current commits are being pulled into master for a version release. xarray pull will occur after the release to provide time to better test that new functionality.

### CODE REVIEW

**Comment by aburrell** [comment]
Pressed the wrong button!

**Comment by aburrell** [comment]
Crap, do you remember how to roll back a commit?

**Comment by rstoneback** [comment]
I would actually need to look that up. About to head into a meeting though.

On Sep 18, 2018, at 2:50 PM, Angeline Burrell <notifications@github.com<mailto:notifications@github.com>> wrote:


Crap, do you remember how to roll back a commit?

**Comment by aburrell** [comment]
I also found a mutli_file_day bug in files, where the routine doesn't consider whether or not you want multiple files per day before removing duplicates.  I fixed this in the model branch.  Would you like me to add the fix here, too?

**Comment by rstoneback** [comment]
Probably so.

For DMSP the time was in the filename as well as the date so I didn’t run into duplicates in this case.

If you are running into this issue then yeah, let’s get it in.

My cosmic binary routine I was thinking of supplying to help with Demeter is buried in a repo at UCAR. I can probably still get it but I saw an initial Demeter support commit from you. Do you still want/need my old code for an example?

Russell

On Sep 21, 2018, at 3:46 PM, Angeline Burrell <notifications@github.com<mailto:notifications@github.com>> wrote:


I also found a mutli_file_day bug in files, where the routine doesn't consider whether or not you want multiple files per day before removing duplicates. I fixed this in the model branch. Would you like me to add the fix here, too?

**Comment by aburrell** [comment]
No, I can successfully read the data now.  I wonder if the file problem is coming from my local list routine again.  The files have h/m/s in the name, so I'm probably not handling that correctly.  You can poke at it if you want, but otherwise I'll take a look on Monday.

File: pysat/instruments/netcdf_pandas.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+
+        
+def init(self):
+    pass
+                
+def load(fnames, tag=None, sat_id=None, sim_multi_file_right=False,
+        sim_multi_file_left=False, root_date = None, **kwargs):
+    """Loads data using pysat.utils.load_netcdf4 .
+
```
Comment (aburrell) [inline pysat/instruments/netcdf_pandas.py]: Parameters should be defined here, and Return values specified.

File: pysat/instruments/netcdf_pandas.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    -------
+    inst = pysat.Instrument('netcdf', 'pandas', custom_kwarg='test')
+    
+    """
+    return pysat.utils.load_netcdf4(fnames, **kwargs)
+
+
+def list_files(tag=None, sat_id=None, data_path=None, format_str=None):
+    """Produce a fake list of files spanning a year"""
+
```
Comment (aburrell) [inline pysat/instruments/netcdf_pandas.py]: Parameters should be defined here, and Return values specified.

File: pysat/instruments/sport_ivm.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+
+        
+def init(self):
+    pass
+                
+def load(fnames, tag=None, sat_id=None, sim_multi_file_right=False,
+        sim_multi_file_left=False, root_date = None, **kwargs):
+    """Loads SPORT IVM data using pysat.utils.load_netcdf4 .
+
```
Comment (aburrell) [inline pysat/instruments/sport_ivm.py]: Parameters should be defined here, and Return values specified.

File: pysat/instruments/sport_ivm.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    inst.load(2019,1)
+    
+    """
+    
+    return pysat.utils.load_netcdf4(fnames, **kwargs)
+
+
+def list_files(tag=None, sat_id=None, data_path=None, format_str=None):
+    """Produce a list of files corresponding to SPORT IVM
+
```
Comment (aburrell) [inline pysat/instruments/sport_ivm.py]: Parameters should be defined here, and Return values specified.  Add "Notes" header before the comments that follow.

File: pysat/instruments/sport_ivm.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        
+    if format_str is None:
+        if tag is '':
+            tag = 'L2'
+        format_str = 'SPORT_'+tag+'_IVM_{year:04d}-{month:02d}-{day:02d}_v{version:02d}r{revision:04d}.NC'
+    return pysat.Files.from_os(data_path=data_path, format_str=format_str)
+
+
+def download(date_array, tag, sat_id, data_path=None, user=None, password=None):
+    print ('Downloads are not currently supported')
```
Comment (aburrell) [inline pysat/instruments/sport_ivm.py]: Add docstring that also states downloads aren't currently supported.

File: pysat/instruments/sport_ivm.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+# dictionary of satellite IDs, list of corresponding tags
+# only one satellite in this case
+sat_ids = {'':['']}
+# good day to download test data for. Downloads aren't currently supported
+test_dates = {'':{'':pysat.datetime(2019,1,1)}}
+
+
+        
+def init(self):
+    pass
```
Comment (aburrell) [inline pysat/instruments/sport_ivm.py]: Is there an acknowledgements statement that this mission will have?

**Comment by rstoneback** [comment]
I implemented all the changes. I also included the nasa_cdaweb_methods and netcdf4_pandas docstrings in the supported instruments area of pysat’s built documentation. I specifically included the list_files, load, download, and init routines to provide a base reference for folks.

Commit has been pushed to develop.

On Sep 24, 2018, at 7:39 AM, Angeline Burrell <notifications@github.com<mailto:notifications@github.com>> wrote:


@aburrell approved this pull request.

All of the changes I suggest improve the docstrings. I recommend making these changes, but didn't spot any problems in the code.

**Pull request merged**